closed/make/ReleaseFile.gmk

```diff
@@ -1,23 +1,33 @@
-# ===========================================================================
-# (c) Copyright IBM Corp. 2017 All Rights Reserved
-# ===========================================================================
-# 
-# This code is free software; you can redistribute it and/or modify it
-# under the terms of the GNU General Public License version 2 only, as
-# published by the Free Software Foundation.  
-#
-# This code is distributed in the hope that it will be useful, but WITHOUT
-# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
-# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
-# version 2 for more details (a copy is included in the LICENSE file that
-# accompanied this code).
-#
-# You should have received a copy of the GNU General Public License version
-# 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
-# 
-# ===========================================================================
+# ===========================================================================
+# (c) Copyright IBM Corp. 2017, 2017 All Rights Reserved
+# ===========================================================================
+# 
+# This code is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 only, as
+# published by the Free Software Foundation.  
+#
+# This code is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+# version 2 for more details (a copy is included in the LICENSE file that
+# accompanied this code).
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+# 
+# ===========================================================================
 
 COMPANY_NAME := Eclipse OpenJ9
 OPENJ9OMR_SHA := $(shell git -C $(OPENJ9OMR_TOPDIR) rev-parse --short HEAD)
 OPENJ9_SHA  := $(shell git -C $(OPENJ9_TOPDIR)  rev-parse --short HEAD)
 SOURCE_REVISION = OpenJDK:$(OPENJDK_SHA) OpenJ9:$(OPENJ9_SHA) OMR:$(OPENJ9OMR_SHA)
+
+define create-info-file
+  $(if $(JDK_ARCH_ABI_PROP_NAME), \
+    $(call info-file-item, "SUN_ARCH_ABI", "$(JDK_ARCH_ABI_PROP_NAME)"))
+  $(call info-file-item, "SOURCE", "$(strip $(SOURCE_REVISION))")
+  $(call info-file-item, "IMPLEMENTOR", "$(COMPANY_NAME)")
+  $(call info-file-item, "OS_NAME", "$(RELEASE_FILE_OS_NAME)")
+  $(call info-file-item, "OS_ARCH", "$(RELEASE_FILE_OS_ARCH)")
+  $(call info-file-item, "JAVA_VERSION", "$(VERSION_MAJOR)")
```

Comment:
As far as I can see this only adds the line that deals with `JAVA_VERSION`: a better approach might be to simply append to the macro `create-info-file`. However, I don't think it will be effective in the definition of `INFO_FILE_VARDEPS` in `make/ReleaseFile.gmk`. Why do we want (or need) `JAVA_VERSION` in that info file?

Reply:
`JAVA_VERSION` in `release` file is required to let JVM determine `J2SE_xx` and `J2SE_HAPE_xx`.
Discussed with @pshipton and @DanHeidinga, will use compile flag to set the Java version and shape values if neither `classlib.properties` nor `release` provides such info hence JCL patch can be avoided.